manager.go

```diff
@@ -220,7 +220,7 @@ func (m *Manager) GoodAddresses(ipversion, pver uint32, services wire.ServiceFla
 				continue
 			}
 		}
-		if pver != 0 && node.ProtocolVersion != pver {
+		if pver != 0 && pver > node.ProtocolVersion {
```

Comment:
I think this should be >= so it's a minimum version.  e.g. You specify you want pver=4 and you get 4+.